### PR TITLE
Fix spawn EINVAL error on Windows

### LINKED ISSUE #83: [BUG] Error: spawn EINVAL
### Issue

I'm playing with the following github action:
```yaml
  prebuild:
    runs-on: ${{ matrix.os }}
    ...
    - name: Setup node
      uses: actions/setup-node@v4
      with:
        node-version: 20
        architecture: ${{ matrix.arch }}
    ...
    - name: Prebuild
      run: prebuildify --target 12.0.0 --napi --strip
```
The main idea is to run **prebuildify** on `macos-latest:x64`, `ubuntu-latest:x64`, `windows-latest:x64` and `windows-latest:x86`.

Everything works fine except for the `windows-latest:x86` where I get the following issue:
```js
node:internal/child_process:414
    throw errnoException(err, 'spawn');
    ^

Error: spawn EINVAL
    at ChildProcess.spawn (node:internal/child_process:414:11)
    at Object.spawn (node:child_process:761:9)
    at D:\a\my-project\node_modules\prebuildify\index.js:230:22
    at D:\a\my-project\node_modules\mkdirp-classic\index.js:30:20
    at FSReqCallback.oncomplete (node:fs:192:23) {
  errno: -4071,
  code: 'EINVAL',
  syscall: 'spawn'
```

### Investigation

The code in `prebuildify index.js:230` is:
```js
var child = proc.spawn(opts.nodeGyp, args, {
  cwd: opts.cwd,
  env: opts.env,
  stdio: opts.quiet ? 'ignore' : 'inherit'
})
```

After debugging I can see that the actual value used for `opts.nodeGyp` is "**node-gyp.cmd**":
```
{
  ...
  stdio: 'inherit',
  args: [
    'node-gyp.cmd',
    'rebuild',
    '--target=12.0.0',
    '--devdir=C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\prebuildify\\node',
    '--arch=ia32',
    '--release'
  ],
  detached: false,
  envPairs: [
    ...,
    ... 98 more items
  ],
  file: 'node-gyp.cmd',
  windowsHide: false,
  windowsVerbatimArguments: false
}
```

### Solution

There seems to have some issues with `spawn` on Windows and most of the time people suggests to use `shell: true` to solve it.

So I monkey patched the previous code in prebuildify like this:
```js
var child = proc.spawn(opts.nodeGyp, args, {
  shell: true,
  cwd: opts.cwd,
  env: opts.env,
  stdio: opts.quiet ? 'ignore' : 'inherit'
})
```
and it indeed solved the issue.

### PR DESCRIPTION
**Pull request by derevnjuk**
fixes #83

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -300,7 +306,7 @@ function npmbin (name) {
 }
 
 function shell () {
-  return os.platform() === 'android' ? 'sh' : undefined
+  return os.platform() === 'android' ? 'sh' : true
```
Comment (vweevers) [inline index.js:309]: We only need it on Windows, on other platforms we can avoid the overhead of a shell.

**Review by vweevers — APPROVED** [review]
LGTM but @mafintosh's comment here makes me unsure: https://github.com/prebuild/prebuildify/pull/81#discussion_r1566043853

**Comment by derevnjuk** [comment]
@vweevers If so, it is better to enforce shell in this particular case like this:

```diff
-var child = execspawn(cmd, {
+var child = proc.spawn(cmd, [], {
  cwd: opts.cwd,
  env: opts.env,
  stdio: 'inherit',
-  shell: opts.shell
+  shell: opts.shell || true,
+  windowsHide: true
})
```

wdyt?

**Comment by vweevers** [comment]
6.0.1

**Pull request merged**